salvo/src/lib/benchmark/base_benchmark.py

```diff
@@ -105,40 +164,32 @@ def run_image(self, image_name: str,
     """
     return self._docker_image.run_image(image_name, run_parameters)
 
-  def pull_images(self) -> List[str]:
-    """Retrieve all images necessary for the benchmark.
+  @abc.abstractmethod
+  def execute_benchmark(self) -> bool:
+    """Run a benchmark
 
-    Retrieve the NightHawk and Envoy images defined in the control
-    object.
+    A class derived from BaseBenchmark is responsible for building and staging
+    the required binaries, tests, and any other artifacts for running a
+    benchmark.
 
-    Returns:
-        a List of required image names that are retrievable. If any image is
-          unavailable, we return an empty list. The intent is for the caller to
-          build the necessary images.
+    For example in the scavenging benchmark, that class must prepare
+    the NightHawk benchamrk and binary docker image, as well as build an Envoy
```

Comment:
Typo: benchamrk.

Reply:
ack.  Will fix.  I need to get spellcheck going for  Salvo :(

Reply:
I am haunted by the same issues as well. Thanks for updating.